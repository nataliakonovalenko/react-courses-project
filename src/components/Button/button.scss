@import "../../styles/variables.scss";

.btn {
  display: inline-block;
  vertical-align: top;
  border: 1px solid $link-color;
  background: $link-color;
  color: $text-color;
  min-width: 150px;
  font-size: 18px;
  line-height: 1.1;
  padding: 20px 15px;
  border-radius: 10px;
  text-transform: uppercase;
  transition: background 0.5s, color 0.5s;
  text-decoration: none;

  &:hover {
    background: darken($link-color, 10%);
  }

  &:disabled {
    cursor: not-allowed;
    pointer-events: none;
    opacity: 0.7;
  }

  &.btn-outline {
    background: transparent;
    color: $link-color;

    &:hover {
      background: $link-color;
      color: $text-color;
    }
  }

  &.btn-success {
    background: rgba($white, 0.1);
    color: $link-color;
    border: none;
    padding: 15px 20px;
    font-size: 20px;

    &:hover {
      background: rgba($white, 0.3);
    }
  }
}

