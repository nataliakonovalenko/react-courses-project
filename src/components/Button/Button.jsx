import React from "react";
import "./button.scss"

export default function Button(props) {
    const {className, ...rest} = props;

    return <button className={className ? `btn ${className}` : "btn" } onClick={props.onButtonClick}>{props.title}</button>
}