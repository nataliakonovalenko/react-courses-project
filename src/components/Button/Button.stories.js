import React from "react";
import { withKnobs, text, boolean } from "@storybook/addon-knobs";
import Button from "./Button";

export default {
    title: "Button",
    component: Button,
    decorators: [withKnobs],
    argTypes: {
        onButtonClick: { action: "clicked" },
    },
};

export const Primary = () => (
    <button className="btn" title="Primary" disabled={boolean('Disabled', false)}>{text('Label', 'Primary button')}</button>
);

export const Secondary = () => (
    <button className="btn btn-outline" title="Secondary" disabled={boolean('Disabled', false)}>{text('Label', 'Secondary button')}</button>
);

export const Success = () => (
    <button className="btn btn-success" title="Success" disabled={boolean('Disabled', false)}>{text('Label', 'Success button')}</button>
);
