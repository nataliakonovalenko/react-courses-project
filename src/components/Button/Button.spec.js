import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Button from "./Button";

describe("snapshot Button component", () => {
    it("should render Button component", () => {
        const component = shallow(<Button />);
        expect(toJson(component)).toMatchSnapshot();
    });
});