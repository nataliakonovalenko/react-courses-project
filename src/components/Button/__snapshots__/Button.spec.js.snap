// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Button component should render Button component 1`] = `
<button
  className="btn"
/>
`;
